libraries/AP_GyroFFT/AP_GyroFFT.cpp

```diff
@@ -248,10 +248,23 @@ void AP_GyroFFT::init(uint16_t loop_rate_hz)
         return;
     }
 
-    const uint8_t harmonics = _ins->get_gyro_harmonic_notch_harmonics();
+    // check for harmonics across all harmonic notch filters
+    // note that we only allow one harmonic notch filter linked to the FFT code
+    uint8_t harmonics = 0;
+    uint8_t num_notches = 0;
+    for (auto &notch : _ins->harmonic_notches) {
+        if (notch.params.enabled() && notch.params.tracking_mode() == HarmonicNotchDynamicMode::UpdateGyroFFT) {
```

Comment:
This should just be ```notch.params.enabled()```

Reply:
sure? If the tracking mode is (for example) 0 for fixed, then why would we look at the number of harmonics?

Reply:
So in theory you need two different checks here. The number of notches should be dependent on the presence of dynamic harmonic, but the harmonics should only be dependent on the notch being enabled. However, this check is already made in AP_InertialSensor::init() and so the only thing you need to check is notch.params.enabled()

Reply:
updated